src/parser.ml

```diff
@@ -302,12 +302,14 @@ let blank buf pos =
   in
   fn `Ini [] (buf, pos) (buf, pos)
 
+let cmdlist_parser : p_cmd loc list Earley.grammar = (parser {l:cmd "."}*)
```

Comment:
Sure, please adapt as you better see fit and I'll rebase, thanks @rlepigre .

> Do you want me to provide a parsing function taking a string as input (instead of a file) as well?

I think I only need the proper `'a grammar` values, the document manager usually wants to implement some form of incremental parsing so it will use some lower-level `Earley` itself.

Note also that I had to make `command` transparent in `Pure`, otherwise I see no way of using it.

I will also submit an issue w.r.t. the interface to `Sign` and `Pure`. Cheers.

Reply:
OK, I applied the changes you need to the `pa` branch. I'll let you merge.